### DISCUSSION TITLE
Github Enterprise server on Ubuntu 22

### DISCUSSION THREAD

**Discussion by Akulkarni-iMX8MP** [community/community#113241] [Enterprise]
I have my own HW for got server and have installed Ubuntu 22 on it. 
I am looking to install Github enterprise server directly on Ubuntu, but when I tried to download the installers I see following options 
1. Installing GitHub Enterprise Server on AWS
2. Installing GitHub Enterprise Server on Azure
3. Installing GitHub Enterprise Server on Google Cloud Platform
4. Installing GitHub Enterprise Server on Hyper-V
5. Installing GitHub Enterprise Server on OpenStack KVM
6. Installing GitHub Enterprise Server on VMware

It looks like none of these can be directly installed on Ubuntu.  It is we can not install directly on Ubuntu we always need to have a virtual machine?
Please help me with clarity, Thanks in advance.

**Comment by ghost** [comment]
> It looks like none of these can be directly installed on Ubuntu. It is we can not install directly on Ubuntu we always need to have a virtual machine? Please help me with clarity, Thanks in advance.

That's correct - you cannot run GitHub Enterprise Server directly on your own OS. In your case, with an Ubuntu 22 server, your best option is to run the KVM version (you don't specifically need OpenStack).

**Reply by DhruvChavda** [reply]
Is there any documentation or reference that I can follow to run this? I want to run it on a digitalocean droplet 💧

**Reply by ghost** [reply]
I am not familiar with digitalocean's offerings, but I am assuming that a droplet is a VM. In that case, you would need to run your own KVM setup on that VM, also assuming that they allow you to run nested virtualization. You could probably follow something like this (https://ubuntu.com/blog/kvm-hyphervisor) but this isn't something I would attempt to run in any production capacity.

**Reply by DhruvChavda** [reply]
Really appreciate it, @ecker-deshaw ! I’ll check this out soon.
Also, @Akulkarni-iMX8MP , would love to hear if you’ve found any alternative for running it on Ubuntu at a production scale.

**Reply by DhruvChavda** [reply]
I've also created a seperate thread for this: https://github.com/orgs/community/discussions/151422
